### PR TITLE
Reviews

### PR DESCRIPTION
**Pull request by agnessa**
Adds the ability to review any project before is is launched, as well as post-launch.

`rake db:migrate`
`rake review:load` (loads questions from db/review_template.json)

### CODE REVIEW

File: app/assets/javascripts/reviews.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+$(document).ready(function(){
+  $('.review-answer').change(function(){
+    var el = $(this);
+    var reviewId = el.data('review-id');
+    var questionId = el.data('question-id');
+    var done = el.val();
```
Comment (lucidstack) [inline app/assets/javascripts/reviews.js]: There seems to be a little problem with the ticking of the answers. Once the checkbox is ticked, the icon never goes back to red thumb down, even if you uncheck the checkbox. 

I think this might be because of the value `'yes'`, that never changes. Perhaps using `el.prop('checked')` here would solve the problem :)

File: app/controllers/review_answers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+class ReviewAnswersController < ApplicationController
+  before_action :authenticate_user!
+
+  respond_to :json
+
+  def create_or_update
+    @review = Review.find(params[:review_id])
+    p = answer_params
+    @answer = @review.answers.where(review_question_id: p[:review_question_id]).first
+    unless @answer
```
Comment (lucidstack) [inline app/controllers/review_answers_controller.rb]: Just a bit of consideration on style: it takes me several brain cycles to understand what `unless/else` does every time, sorry :grin: If you could swap them over, I think it would make it more straightforward to read. 

Alternatively, `@answer = @review.answers.create_with(p).find_or_create_by(review_question_id: p[:review_question_id])` could be used to avoid the conditional block

File: app/controllers/review_answers_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  def show
+  end
+
+  private
+
+  def answer_params
+    params.require(:answer).permit(:done, :review_question_id).tap do |p|
+      case p[:done]
```
Comment (lucidstack) [inline app/controllers/review_answers_controller.rb]: can this be just a `if p[:done] == 'yes'/else` block?

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  def set_sections
+    @sections = ReviewSection.order(:sort_order)
+    @sections = if @review.nil? || @review.new_record?
+      @sections.includes(:questions)
+    else
+      @sections.includes(:questions => :answers)
+    end
+    @sections = @sections.references(:questions).
+    order('review_sections.sort_order, review_questions.sort_order')
```
Comment (lucidstack) [inline app/controllers/reviews_controller.rb]: A level of indentation more here would make the call to `order` clearer! :smiley:

File: app/helpers/projects_helper.rb [hunk 5]
```diff
@@ -19,4 +19,25 @@ def display_text project, field, display_text=nil
     content_tag(:p, content_tag(:strong, (display || field.to_s.titleize)+":")) +
       content_tag(:div, simple_format(h project.send(field)), class: "block-of-text")
   end
+
+  def review_score(review)
+    return '' unless review
+    link_to review_path(review) do
+      if review.result == 1.0
```
Comment (lucidstack) [inline app/helpers/projects_helper.rb]: `1.0` here could go in a constant, perhaps just above the method definition

File: app/helpers/reviews_helper.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      end
+    else
+      review_checkbox(question, 'yes', answer.try(:done?), disabled)
+    end
+  end
+
+  def feedback(question)
+    answer = @review.try(:answer, question)
+    feedback = answer && answer.is_acceptable?(question) ? 'success' : 'error'
+    content_tag(:div, class: "col-sm-2 feedback #{feedback}") do
```
Comment (lucidstack) [inline app/helpers/reviews_helper.rb]: Is there any chance you could extract some of these blocks in private methods, or lambdas? It's a bit hard to follow, unfortunately. :confused:

**Comment by timwilki** [comment]
For the list items that can be rectified in labs, perhaps it's worth adding a shortcut to the editing project page next to that item. This way a user doesn't have to go to projects > search projects > etc.

**Comment by agnessa** [comment]
Thank you guys, I have fixed the issues you found.
